packages/test/test-end-to-end-tests/src/test/gcDataStoreRequests.spec.ts

```diff
@@ -66,51 +68,85 @@ describeNoCompat("GC Data Store Requests", (getTestObjectProvider) => {
+                    sequenceNumber = op.sequenceNumber;
+                    resolve(true);
+                    directory.off("op", listener);
+                }
+            };
+            directory.on("op", listener);
+        });
+        assert(sequenceNumber !== undefined);
+        return sequenceNumber;
     }
 
+    const createContainer = async (): Promise<IContainer> => provider.createContainer(runtimeFactory);
+    const loadContainer = async (summaryVersion: string): Promise<IContainer> => {
+        const requestHeader = {
+            [LoaderHeader.version]: summaryVersion,
+        };
+        return provider.loadContainer(runtimeFactory, undefined /* options */, requestHeader);
+    };
+
     beforeEach(async () => {
         provider = getTestObjectProvider();
 
         // Create a Container for the first client.
-        container1 = await createContainer();
+        mainContainer = await createContainer();
+
+        // Create and setup a summary collection that will be used to track and wait for summaries.
+        summaryCollection = new SummaryCollection(
+            mainContainer.deltaManager.initialSequenceNumber,
+            new TelemetryNullLogger(),
+        );
+        mainContainer.deltaManager.inbound.on("op",
+            (op) => summaryCollection.handleOp(op));
     });
 
     it("should fail requests with externalRequest flag for unreferenced data stores", async () => {
-        const dataStore1 = await requestFluidObject<TestDataObject>(container1, "default");
+        const directoryKey = "dataStore2";
+        const dataStore1 = await requestFluidObject<TestDataObject>(mainContainer, "default");
 
         // Create a second data store (dataStore2) and add its handle to mark it as referenced.
         const dataStore2 = await factory.createInstance(dataStore1._context.containerRuntime);
-        dataStore1._root.set("dataStore2", dataStore2.handle);
+        dataStore1._root.set(directoryKey, dataStore2.handle);
 
-        // Wait for ops to be processed so that summarizer creates dataStore2.
-        await provider.ensureSynchronized();
+        // Wait for the set to be processed. Then wait for a summary that includes the sequence number of the above op.
+        let sequenceNumber = await waitForDirectoryOp(dataStore1._root, directoryKey);
+        await waitForSummary(sequenceNumber);
 
         // Now delete the handle so that dataStore2 is marked as unreferenced.
-        dataStore1._root.delete("dataStore2");
+        dataStore1._root.delete(directoryKey);
 
-        // Wait for the summarizer to generate a summary where dataStore2 is unreferenced.
-        await waitForSummary(container1);
+        // Wait for the delete to be processed. Then wait for a summary that includes the sequence number of the above
+        // op. Also, get this summary's version so that we can load a new container with it.
+        sequenceNumber = await waitForDirectoryOp(dataStore1._root, directoryKey);
```

Comment:
instead of listing to the directory op, may be we can just ensure synchronized and the check the sequenceNumber at that point?  At the same time, listen to summary op and keep track of summary that got generated and find/wait the one that includ this sequenceNumber?

Reply:
The problem is that by the time ensureSynchronized completes, the summaryOp may have been sent out and processed. So, the last seq# would include summary op's seq# and we will not get an ack for this.

Reply:
i call out the same thing above. use the saved event on container, then check the delta mangers last seq

Reply:
Updated the tests to wait for `ensureSynchronized` instead. However, I am still worried that while we are for `ensureSynchronized` to happen, the summary op may be send and received, updating the lastSequenceNumber.

Reply:
Would it be better for force a summary instead of waiting for the summary heuristic to trigger?  IMO, unless you are testing for heuristic, actively do thing that is needed is better.

Reply:
We don't have that capability in the summaizer today. We can call `summarize` on the container runtime directly but that doesn't involve the summarizer. We may want to add this option to the summarizer and then these tests become a lot simpler :-).
I have talked to Arin about this as part of our long running summary tests tasks. I will create and issue and talk to him again.